pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -28,9 +28,14 @@ let
     ghostscript = ghostscript_headless;
   };
 
+  tlpdb = import ./tlpdb.nix;
+
+  tlpdbVersion = tlpdb."00texlive.config";
+
   # the set of TeX Live packages, collections, and schemes; using upstream naming
-  tl = let
-    orig = import ./tlpdb.nix;
+  tl = assert lib.assertMsg (tlpdbVersion.year == version.texliveYear) "TeX Live year in tlpdb.nix and texlive do not match, refusing to evaluate";
+    assert lib.assertMsg (tlpdbVersion.frozen == version.final) "TeX Live final status in tlpdb.nix and texlive do not match, refusing to evaluate"; let
```

Comment:
Nix is lazy, but frustratingly not lazy enough: this assert makes it impossible to evaluate `texlive.tlpdb-nix`, even though it is independent of `tl`. Is there a better place for this assert?

Reply:
Oof, I forgot that `texlive = tl / ...` so evaluating `texlive` itself calls the assert. Fixed.